<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>FLOW-MR-tutorial • FLOWMR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="FLOW-MR-tutorial">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">FLOWMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/FLOW-MR-tutorial.html">Get Started</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>FLOW-MR-tutorial</h1>
            
      

      <div class="d-none name"><code>FLOW-MR-tutorial.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>FLOW-MR</code> is an R package for performing Mendelian
Randomization under the mediation setting. This tutorial demonstrates
how to use the package step by step.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install devtools if not installed</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The downloaded binary packages are in</span></span>
<span><span class="co">#&gt;  /var/folders/1p/kyg0zmtx4gdg27q_5x4d78fc0000gn/T//RtmpYVc0Jt/downloaded_packages</span></span>
<span></span>
<span><span class="co"># Install FLOW-MR</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ZixuanWu1/FLOW-MR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; rlang        (1.1.4    -&gt; 1.1.5   ) [CRAN]</span></span>
<span><span class="co">#&gt; cli          (3.6.3    -&gt; 3.6.4   ) [CRAN]</span></span>
<span><span class="co">#&gt; pillar       (1.9.0    -&gt; 1.10.1  ) [CRAN]</span></span>
<span><span class="co">#&gt; R6           (2.5.1    -&gt; 2.6.1   ) [CRAN]</span></span>
<span><span class="co">#&gt; Rcpp         (1.0.13   -&gt; 1.0.14  ) [CRAN]</span></span>
<span><span class="co">#&gt; withr        (3.0.1    -&gt; 3.0.2   ) [CRAN]</span></span>
<span><span class="co">#&gt; gtable       (0.3.5    -&gt; 0.3.6   ) [CRAN]</span></span>
<span><span class="co">#&gt; RcppArmad... (14.0.2-1 -&gt; 14.2.3-1) [CRAN]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The downloaded binary packages are in</span></span>
<span><span class="co">#&gt;  /var/folders/1p/kyg0zmtx4gdg27q_5x4d78fc0000gn/T//RtmpYVc0Jt/downloaded_packages</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="color: #00BBBB;">R CMD build</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; * checking for file ‘/private/var/folders/1p/kyg0zmtx4gdg27q_5x4d78fc0000gn/T/RtmpYVc0Jt/remotes6e32145d8194/ZixuanWu1-FLOW-MR-2165c20/DESCRIPTION’ ... OK</span></span>
<span><span class="co">#&gt; * preparing ‘FLOWMR’:</span></span>
<span><span class="co">#&gt; * checking DESCRIPTION meta-information ... OK</span></span>
<span><span class="co">#&gt; * cleaning src</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:48:21</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:49:22</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:50:26</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:52:27</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:54:29</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:56:30</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:58:17</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:59:18</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:64:9</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:65:9</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:66:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:67:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:68:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:69:13</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:24:12</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:25:16</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:26:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:27:13</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:28:12</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:29:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:30:11</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:31:13</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at zero.centered.em.Rd:42:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at zero.centered.em.Rd:43:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at zero.centered.em.Rd:44:10</span></span>
<span><span class="co">#&gt; * installing the package to process help pages</span></span>
<span><span class="co">#&gt; * saving partial Rd database</span></span>
<span><span class="co">#&gt; * cleaning src</span></span>
<span><span class="co">#&gt; * checking for LF line-endings in source and make files and shell scripts</span></span>
<span><span class="co">#&gt; * checking for empty or unneeded directories</span></span>
<span><span class="co">#&gt; * building ‘FLOWMR_1.0.tar.gz’</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-package">Load the Package<a class="anchor" aria-label="anchor" href="#load-the-package"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">FLOWMR</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-usage">Example Usage<a class="anchor" aria-label="anchor" href="#example-usage"></a>
</h2>
<div class="section level3">
<h3 id="step-1-prepare-input-data">Step 1: Prepare Input Data<a class="anchor" aria-label="anchor" href="#step-1-prepare-input-data"></a>
</h3>
<p>Here, we use <code>GRAPPLE</code> to preprocess the input data. We
can download the <code>GRAPPLE</code> package using the following
command:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"jingshuw/grapple"</span><span class="op">)</span></span></code></pre></div>
<p>Next, we use <code>GRAPPLE</code> to preprocess the data following a
three-sample Mendelian Randomization (MR) design. The sel.file is used
to select genome-wide significant SNPs, the exp.file contains the
exposures of interest, and the out.file contains the outcomes of
interest. For more details, visit <a href="https://github.com/jingshuw/GRAPPLE" class="external-link">GRAPPLE</a> on GitHub.</p>
<p>In this example we use adult BMI from <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files" class="external-link">GIANT</a>
and childhoood BMI from <a href="http://egg-consortium.org" class="external-link">EGG</a> as
selections files. We use childhood body size and bmi adult from <a href="https://www.nealelab.is/uk-biobank" class="external-link">UK biobank</a> as exposures,
and Breast Cancer from this <a href="https://pubmed.ncbi.nlm.nih.gov/29059683" class="external-link">paper</a> as
outcome.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GRAPPLE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Selection file of snps</span></span>
<span><span class="va">sel.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"~/Downloads/research/BMI-giant17eu.csv"</span>, <span class="st">"~/Downloads/research/bmi_child_egg.csv"</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># Exposure file</span></span>
<span><span class="va">exp.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"~/Downloads/research/childhood_bmi_nat_comm_2019_8years"</span>,<span class="st">"~/Downloads/research/BMI_adult.csv"</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># Outcome file</span></span>
<span><span class="va">out.file</span> <span class="op">&lt;-</span> <span class="st">"~/Downloads/research/Breast-Micha17erp.csv"</span></span>
<span></span>
<span><span class="co"># Use plink to select independent significant SNPs</span></span>
<span><span class="va">plink_refdat</span> <span class="op">&lt;-</span> <span class="st">"~/Downloads/research/data_maf0.01_rs_ref/data_maf0.01_rs_ref"</span></span>
<span></span>
<span><span class="co">## Use max.p.thres to decide the significance level of SNPs</span></span>
<span><span class="co">## Use cal.cor = T to compute the noise correlation</span></span>
<span><span class="va">data.list</span> <span class="op">&lt;-</span> <span class="fu">GRAPPLE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GRAPPLE/man/getInput.html" class="external-link">getInput</a></span><span class="op">(</span><span class="va">sel.file</span>, <span class="va">exp.file</span>, <span class="va">out.file</span>, <span class="va">plink_refdat</span>, max.p.thres <span class="op">=</span><span class="fl">1e-4</span>,</span>
<span>                               plink_exe <span class="op">=</span> <span class="st">"~/Downloads/research/plink_mac_20210606/plink"</span>, cal.cor <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Marker candidates will not be obtained as number of risk factors k &gt; 1"</span></span>
<span><span class="co">#&gt; [1] "loading data for selection: ~/Downloads/research/BMI-giant17eu.csv ..."</span></span>
<span><span class="co">#&gt; [1] "loading data for selection: ~/Downloads/research/bmi_child_egg.csv ..."</span></span>
<span><span class="co">#&gt; [1] "loading data from exposure: ~/Downloads/research/childhood_bmi_nat_comm_2019_8years ..."</span></span>
<span><span class="co">#&gt; [1] "loading data from exposure: ~/Downloads/research/BMI_adult.csv ..."</span></span>
<span><span class="co">#&gt; [1] "loading data from outcome: ~/Downloads/research/Breast-Micha17erp.csv ..."</span></span>
<span><span class="co">#&gt; [1] "Start clumping using PLINK ..."</span></span>
<span><span class="co">#&gt; [1] "262 independent genetic instruments extracted. Done!"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-2-run-flow-mr">Step 2: Run FLOW-MR<a class="anchor" aria-label="anchor" href="#step-2-run-flow-mr"></a>
</h3>
<p>To run FLOW-MR, we need to prepare two input files,
<code>Gamma_hat</code> and <code>Sd_hat</code>. Also ensuring that the
time order is reversed:</p>
<p>Here, you can optionally provide <code>cor_mat</code>, a K by K
matrix which represents the shared noise correlation between GWAS
summary statistics across traits. This is particularly useful when the
GWAS summary statistics include overlapping samples.
<code>GRAPPLE</code> estimates the noise correlation by first selecting
non-significant SNPs (e.g., p-value &gt; 0.5) and then computing the
correlation between the estimated effect sizes. The default value is a
diagonal matrix.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span><span class="va">data.list</span><span class="op">$</span><span class="va">data</span>;</span>
<span></span>
<span><span class="co"># Run the mediation method</span></span>
<span><span class="va">Gamma_hat</span> <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">gamma_out1</span>, </span>
<span>                 <span class="va">dat</span><span class="op">$</span><span class="va">gamma_exp2</span>,</span>
<span>                 <span class="va">dat</span><span class="op">$</span><span class="va">gamma_exp1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Sd_hat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">se_out1</span>, </span>
<span>               <span class="va">dat</span><span class="op">$</span><span class="va">se_exp2</span>,</span>
<span>               <span class="va">dat</span><span class="op">$</span><span class="va">se_exp1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cor_mat</span> <span class="op">=</span> <span class="va">data.list</span><span class="op">$</span><span class="va">cor.mat</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">:</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="va">result</span> <span class="op">=</span> <span class="fu"><a href="../reference/BayesMediation.html">BayesMediation</a></span><span class="op">(</span><span class="va">Gamma_hat</span>, <span class="va">Sd_hat</span>, cor <span class="op">=</span> <span class="va">cor_mat</span> <span class="op">)</span></span>
<span><span class="co">#&gt; [1] "2025-03-14 09:59:59 CDT"</span></span>
<span><span class="co">#&gt; [1] "2025-03-14 10:00:33 CDT"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-look-at-summary-of-direct-effects">Step 3: Look at summary of direct effects<a class="anchor" aria-label="anchor" href="#step-3-look-at-summary-of-direct-effects"></a>
</h3>
<p>In this step, we print the summary of direct effects, where each row
corresponds to a parameter.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">summary</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   mean          var           sd         2.5%          50%</span></span>
<span><span class="co">#&gt; B[1,2]    -0.003799960 1.499704e-02 0.1224623953 -0.242506035 -0.004762849</span></span>
<span><span class="co">#&gt; B[1,3]    -0.208916503 6.275352e-03 0.0792171177 -0.373170259 -0.207991179</span></span>
<span><span class="co">#&gt; B[2,3]     0.665471334 8.893716e-03 0.0943064996  0.501847350  0.659909168</span></span>
<span><span class="co">#&gt; sigma      0.421958927 1.197907e-02 0.1094489416  0.267911905  0.402421742</span></span>
<span><span class="co">#&gt; sigma1[1]  0.020422056 1.224979e-05 0.0034999697  0.014939265  0.019985341</span></span>
<span><span class="co">#&gt; sigma1[2]  0.027266894 7.479270e-05 0.0086482775  0.016470575  0.025459393</span></span>
<span><span class="co">#&gt; sigma1[3]  0.051531506 1.647916e-04 0.0128371172  0.032076821  0.049623151</span></span>
<span><span class="co">#&gt; sigma0[1]  0.004014919 9.450237e-07 0.0009721233  0.002465223  0.003908847</span></span>
<span><span class="co">#&gt; sigma0[2]  0.010903923 1.485353e-06 0.0012187504  0.008371434  0.010932066</span></span>
<span><span class="co">#&gt; sigma0[3]  0.015815887 3.905375e-06 0.0019762020  0.012244022  0.015725888</span></span>
<span><span class="co">#&gt; p[1]       0.201621997 4.761344e-03 0.0690024897  0.085769722  0.196063901</span></span>
<span><span class="co">#&gt; p[2]       0.034691299 8.198708e-04 0.0286333866  0.003505214  0.026866603</span></span>
<span><span class="co">#&gt; p[3]       0.045608505 4.713271e-04 0.0217100695  0.012394625  0.042635409</span></span>
<span><span class="co">#&gt;                  97.5%        ESS     Rhat</span></span>
<span><span class="co">#&gt; B[1,2]     0.238086832  695.66401 1.002933</span></span>
<span><span class="co">#&gt; B[1,3]    -0.057080287  565.43372 1.005385</span></span>
<span><span class="co">#&gt; B[2,3]     0.856221073   56.94984 1.000525</span></span>
<span><span class="co">#&gt; sigma      0.689571449 4427.07302 1.004792</span></span>
<span><span class="co">#&gt; sigma1[1]  0.028565783 1249.14928 1.002086</span></span>
<span><span class="co">#&gt; sigma1[2]  0.049209360 3703.62082 1.000397</span></span>
<span><span class="co">#&gt; sigma1[3]  0.082478386 2669.62396 1.000203</span></span>
<span><span class="co">#&gt; sigma0[1]  0.006192147  261.61314 1.004128</span></span>
<span><span class="co">#&gt; sigma0[2]  0.013243683  487.88586 1.000888</span></span>
<span><span class="co">#&gt; sigma0[3]  0.020015323  640.43459 1.001528</span></span>
<span><span class="co">#&gt; p[1]       0.354793861  616.29918 1.009205</span></span>
<span><span class="co">#&gt; p[2]       0.109812666  878.61227 1.001145</span></span>
<span><span class="co">#&gt; p[3]       0.095434215 1667.64568 1.000631</span></span></code></pre></div>
<p>In this summary, we provide the estimated posterior mean, variance,
standard deviation, 2.5% quantile, median, 97.5% quantile, effective
sample size (ESS), and Gelman-Rubin Rhat statistic for each parameter.
The ESS quantifies the number of independent samples, while Rhat (the
Gelman-Rubin statistic) assesses convergence. See <a href="https://mc-stan.org/docs/2_18/reference-manual/effective-sample-size-section.html" class="external-link">ESS</a>
and <a href="https://mc-stan.org/rstan/reference/Rhat.html" class="external-link">Rhat</a> for
a detailed explanations of ESS and Rhat.</p>
<p>For the meaning of each parameter, <code>B[l, k]</code> represents
the effect of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>-th
latest trait on the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>-th
latest trait. For example: - <code>B[1, 2]</code> represents the effect
of adult BMI on breast cancer, - <code>(B[1, 3]</code> represents the
direct effect of childhood BMI on breast cancer, - <code>(B[2, 3]</code>
represents the direct effect of childhood BMI on adult BMI.</p>
<p>For the meaning of other parameters, we assume the following priors
in our manuscript:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>α</mi><mrow><mi>k</mi><mi>j</mi></mrow></msub><mo>∼</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><msub><mi>p</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mi>N</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><msubsup><mi>σ</mi><mrow><mi>k</mi><mn>0</mn></mrow><mn>2</mn></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>p</mi><mi>k</mi></msub><mi>N</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><msubsup><mi>σ</mi><mrow><mi>k</mi><mn>1</mn></mrow><mn>2</mn></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">
\alpha_{kj} \sim (1 - p_k)N(0, \sigma_{k0}^2) + p_k N(0, \sigma_{k1}^2)
</annotation></semantics></math></p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>β</mi><mrow><mi>k</mi><mi>l</mi></mrow></msub><mo>∼</mo><mi>N</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><msup><mi>σ</mi><mn>2</mn></msup><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mspace width="1.0em"></mspace><mi>l</mi><mo>&gt;</mo><mi>k</mi></mrow><annotation encoding="application/x-tex">
\beta_{kl} \sim N(0, \sigma^2), \quad l &gt; k
</annotation></semantics></math></p>
<p>We also include the posterior summary of these parameters. In the
summary: - <code>sigma1[k]</code> represents the standard deviation of
the slab component for trait
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
- <code>sigma0[k]</code> represents the standard deviation of the spike
component for trait
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
- <code>p[k]</code> represents the proportion of the slab component, and
- <code>\sigma</code> represents the standard deviation of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>B</mi><annotation encoding="application/x-tex">B</annotation></semantics></math>.</p>
<p>The effects should be considered as significant if its credible
interval does not include 0.</p>
<p>If non-convergence issues arise (e.g., a large Rhat value), you can
generate a trace plot of the posterior samples using the following
command:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">FLOWMR</span><span class="fu">::</span><span class="fu"><a href="../reference/traceplot.html">traceplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"B"</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="step-4">Step 4<a class="anchor" aria-label="anchor" href="#step-4"></a>
</h3>
<p>To estimate path-wise effects, use the following command. For
example, the code below evaluates the indirect effect of childhood BMI
on breast cancer through adult BMI:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path_effect</span> <span class="op">=</span> <span class="fu"><a href="../reference/indirect_effect.html">indirect_effect</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, K <span class="op">=</span> <span class="fl">3</span>, path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">1</span><span class="op">)</span>,  warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">path_effect</span><span class="op">)</span></span>
<span><span class="co">#&gt;            Mean       2.5%          50%    97.5%</span></span>
<span><span class="co">#&gt; 1 -7.288468e-05 -0.1631804 -0.003021919 0.177939</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This tutorial introduced the key functionalities of FLOW-MR. For more
details, refer to the package documentation.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zixuan Wu, Ethan Lewis, Qingyuan Zhao, Jingshu Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
