<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>FLOW-MR-tutorial • FLOWMR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="FLOW-MR-tutorial">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">FLOWMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/FLOW-MR-tutorial.html">Get Started</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>FLOW-MR-tutorial</h1>
            
      

      <div class="d-none name"><code>FLOW-MR-tutorial.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>FLOW-MR</code> is an R package for performing Mendelian
Randomization under the mediation setting. This tutorial demonstrates
how to use the package step by step.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install devtools if not installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span></span>
<span><span class="co"># Install FLOW-MR</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ZixuanWu1/FLOW-MR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; rlang        (1.1.4    -&gt; 1.1.5   ) [CRAN]</span></span>
<span><span class="co">#&gt; cli          (3.6.3    -&gt; 3.6.4   ) [CRAN]</span></span>
<span><span class="co">#&gt; pillar       (1.9.0    -&gt; 1.10.1  ) [CRAN]</span></span>
<span><span class="co">#&gt; R6           (2.5.1    -&gt; 2.6.1   ) [CRAN]</span></span>
<span><span class="co">#&gt; Rcpp         (1.0.13   -&gt; 1.0.14  ) [CRAN]</span></span>
<span><span class="co">#&gt; withr        (3.0.1    -&gt; 3.0.2   ) [CRAN]</span></span>
<span><span class="co">#&gt; gtable       (0.3.5    -&gt; 0.3.6   ) [CRAN]</span></span>
<span><span class="co">#&gt; RcppArmad... (14.0.2-1 -&gt; 14.2.3-1) [CRAN]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; The downloaded binary packages are in</span></span>
<span><span class="co">#&gt;  /var/folders/1p/kyg0zmtx4gdg27q_5x4d78fc0000gn/T//RtmpL9P6lU/downloaded_packages</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="color: #00BBBB;">R CMD build</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; * checking for file ‘/private/var/folders/1p/kyg0zmtx4gdg27q_5x4d78fc0000gn/T/RtmpL9P6lU/remotes17574441c9acc/ZixuanWu1-FLOW-MR-888679f/DESCRIPTION’ ... OK</span></span>
<span><span class="co">#&gt; * preparing ‘FLOWMR’:</span></span>
<span><span class="co">#&gt; * checking DESCRIPTION meta-information ... OK</span></span>
<span><span class="co">#&gt; * cleaning src</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:48:21</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:49:22</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:50:26</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:52:27</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:54:29</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:56:30</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:58:17</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at BayesMediation.Rd:59:18</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:57:21</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:58:22</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:59:26</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:61:27</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:63:29</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:65:30</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:67:17</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at MultivariateMediation.Rd:68:18</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:64:9</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:65:9</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:66:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:67:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:68:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper.Rd:69:13</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper_multivariate.Rd:71:9</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper_multivariate.Rd:72:9</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper_multivariate.Rd:73:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper_multivariate.Rd:74:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper_multivariate.Rd:75:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at gibbs_wrapper_multivariate.Rd:76:13</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:24:12</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:25:16</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:26:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:27:13</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:28:12</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:29:14</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:30:11</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at summary_gibbs.Rd:31:13</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at zero.centered.em.Rd:42:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at zero.centered.em.Rd:43:10</span></span>
<span><span class="co">#&gt; Warning: bad markup (extra space?) at zero.centered.em.Rd:44:10</span></span>
<span><span class="co">#&gt; * installing the package to process help pages</span></span>
<span><span class="co">#&gt; * saving partial Rd database</span></span>
<span><span class="co">#&gt; * cleaning src</span></span>
<span><span class="co">#&gt; * checking for LF line-endings in source and make files and shell scripts</span></span>
<span><span class="co">#&gt; * checking for empty or unneeded directories</span></span>
<span><span class="co">#&gt; * building ‘FLOWMR_1.0.tar.gz’</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-the-package">Load the Package<a class="anchor" aria-label="anchor" href="#load-the-package"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">FLOWMR</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-usage">Example Usage<a class="anchor" aria-label="anchor" href="#example-usage"></a>
</h2>
<div class="section level3">
<h3 id="step-1-prepare-input-data">Step 1: Prepare Input Data<a class="anchor" aria-label="anchor" href="#step-1-prepare-input-data"></a>
</h3>
<p>Here, we use <code>GRAPPLE</code> to preprocess the input data. We
can download the <code>GRAPPLE</code> package using the following
command:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"jingshuw/grapple"</span><span class="op">)</span></span></code></pre></div>
<p>Next, we use <code>GRAPPLE</code> to preprocess the data following a
three-sample Mendelian Randomization (MR) design. The sel.file is used
to select genome-wide significant SNPs, the exp.file contains the
exposures of interest, and the out.file contains the outcomes of
interest. For more details, visit <a href="https://github.com/jingshuw/GRAPPLE" class="external-link">GRAPPLE</a> on GitHub.</p>
<p>In this example we use adult BMI from <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files" class="external-link">GIANT</a>
and childhoood BMI from <a href="http://egg-consortium.org" class="external-link">EGG</a> as
selections files. We use childhood body BMI from <a href="https://www.fhi.no/en/ch/studies/moba/for-forskere-artikler/gwas-data-from-moba/" class="external-link">MOBA</a>,
adult BMI from <a href="https://www.nealelab.is/uk-biobank" class="external-link">UK
biobank</a> as exposures, and Breast Cancer from this <a href="https://pubmed.ncbi.nlm.nih.gov/29059683" class="external-link">paper</a> as outcome.
One can download the traits from this <a href="https://www.dropbox.com/scl/fo/mq6zge2zy83a7jzhsiyqb/AGNJ31iDKQHcKk0xQ5JrbpU?rlkey=fnbggfm1gubwxunuw77ns81tb&amp;st=m8dcrh92&amp;dl=0" class="external-link">link</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GRAPPLE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Selection file of snps</span></span>
<span><span class="va">sel.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BMI-giant17eu.csv"</span>, <span class="st">"BMIchild-egg15.csv"</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># Exposure file</span></span>
<span><span class="va">exp.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"BMI8year-moba19"</span>,<span class="st">"BMIadult-ukb.csv"</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># Outcome file</span></span>
<span><span class="va">out.file</span> <span class="op">&lt;-</span> <span class="st">"Breast-Micha17erp.csv"</span></span>
<span></span>
<span><span class="co"># Use plink to select independent significant SNPs</span></span>
<span><span class="va">plink_refdat</span> <span class="op">&lt;-</span> <span class="st">"data_maf0.01_rs_ref/data_maf0.01_rs_ref"</span></span>
<span></span>
<span><span class="co">## Use max.p.thres to decide the significance level of SNPs</span></span>
<span><span class="co">## Use cal.cor = T to compute the noise correlation</span></span>
<span><span class="va">data.list</span> <span class="op">&lt;-</span> <span class="fu">GRAPPLE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GRAPPLE/man/getInput.html" class="external-link">getInput</a></span><span class="op">(</span><span class="va">sel.file</span>, <span class="va">exp.file</span>, <span class="va">out.file</span>, <span class="va">plink_refdat</span>, max.p.thres <span class="op">=</span><span class="fl">1e-3</span>,</span>
<span>                               plink_exe <span class="op">=</span> <span class="st">"plink_mac_20210606/plink"</span>, cal.cor <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Marker candidates will not be obtained as number of risk factors k &gt; 1"</span></span>
<span><span class="co">#&gt; [1] "loading data for selection: BMI-giant17eu.csv ..."</span></span>
<span><span class="co">#&gt; [1] "loading data for selection: BMIchild-egg15.csv ..."</span></span>
<span><span class="co">#&gt; [1] "loading data from exposure: BMI8year-moba19 ..."</span></span>
<span><span class="co">#&gt; [1] "loading data from exposure: BMIadult-ukb.csv ..."</span></span>
<span><span class="co">#&gt; [1] "loading data from outcome: Breast-Micha17erp.csv ..."</span></span>
<span><span class="co">#&gt; [1] "Start clumping using PLINK ..."</span></span>
<span><span class="co">#&gt; [1] "487 independent genetic instruments extracted. Done!"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-2-run-flow-mr">Step 2: Run FLOW-MR<a class="anchor" aria-label="anchor" href="#step-2-run-flow-mr"></a>
</h3>
<p>To run FLOW-MR, we need to prepare two input files,
<code>Gamma_hat</code> and <code>Sd_hat</code>. Also ensuring that the
time order is reversed. Then the function <code>BayesMediation</code>
will execute the Gibbs sampler.</p>
<p>Here, you can optionally provide <code>cor_mat</code>, a K by K
matrix which represents the shared noise correlation between GWAS
summary statistics across traits. This is particularly useful when the
GWAS summary statistics include overlapping samples.
<code>GRAPPLE</code> estimates the noise correlation by first selecting
non-significant SNPs (e.g., p-value &gt; 0.5) and then computing the
correlation between the estimated effect sizes. The default value is a
diagonal matrix.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span><span class="va">data.list</span><span class="op">$</span><span class="va">data</span>;</span>
<span></span>
<span><span class="co"># Run the mediation method</span></span>
<span><span class="va">Gamma_hat</span> <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">gamma_out1</span>, </span>
<span>                 <span class="va">dat</span><span class="op">$</span><span class="va">gamma_exp2</span>,</span>
<span>                 <span class="va">dat</span><span class="op">$</span><span class="va">gamma_exp1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Sd_hat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">dat</span><span class="op">$</span><span class="va">se_out1</span>, </span>
<span>               <span class="va">dat</span><span class="op">$</span><span class="va">se_exp2</span>,</span>
<span>               <span class="va">dat</span><span class="op">$</span><span class="va">se_exp1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cor_mat</span> <span class="op">=</span> <span class="va">data.list</span><span class="op">$</span><span class="va">cor.mat</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">:</span><span class="fl">1</span><span class="op">]</span></span>
<span></span>
<span><span class="va">result</span> <span class="op">=</span> <span class="fu"><a href="../reference/BayesMediation.html">BayesMediation</a></span><span class="op">(</span><span class="va">Gamma_hat</span>, <span class="va">Sd_hat</span>, cor <span class="op">=</span> <span class="va">cor_mat</span>, inv <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "2025-03-17 17:48:54 CDT"</span></span>
<span><span class="co">#&gt; [1] "2025-03-17 17:50:38 CDT"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-look-at-summary-of-direct-effects">Step 3: Look at summary of direct effects<a class="anchor" aria-label="anchor" href="#step-3-look-at-summary-of-direct-effects"></a>
</h3>
<p>In this step, we print the summary of direct effects, where each row
corresponds to a parameter.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">summary</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   mean          var           sd         2.5%          50%</span></span>
<span><span class="co">#&gt; B[1,2]     0.158571124 1.708075e-02 0.1306933512 -0.088594706  0.155708787</span></span>
<span><span class="co">#&gt; B[1,3]    -0.416820534 3.037425e-02 0.1742821007 -0.772970801 -0.408581119</span></span>
<span><span class="co">#&gt; B[2,3]     0.729903282 8.361712e-03 0.0914423953  0.564416591  0.726111975</span></span>
<span><span class="co">#&gt; sigma      0.432940868 1.258926e-02 0.1122018645  0.273914991  0.412901419</span></span>
<span><span class="co">#&gt; sigma1[1]  0.025897252 1.995543e-05 0.0044671501  0.018968703  0.025298456</span></span>
<span><span class="co">#&gt; sigma1[2]  0.025400411 5.691362e-05 0.0075441115  0.015802744  0.023753648</span></span>
<span><span class="co">#&gt; sigma1[3]  0.046966628 9.573373e-05 0.0097843616  0.032002343  0.045542140</span></span>
<span><span class="co">#&gt; sigma0[1]  0.004279803 7.322883e-07 0.0008557385  0.002882453  0.004194274</span></span>
<span><span class="co">#&gt; sigma0[2]  0.008336840 7.043863e-07 0.0008392773  0.006698575  0.008356154</span></span>
<span><span class="co">#&gt; sigma0[3]  0.010865672 1.488539e-06 0.0012200570  0.008584134  0.010815020</span></span>
<span><span class="co">#&gt; p[1]       0.096893667 1.223057e-03 0.0349722279  0.039493403  0.093387522</span></span>
<span><span class="co">#&gt; p[2]       0.023916549 3.255310e-04 0.0180424783  0.002547653  0.019647269</span></span>
<span><span class="co">#&gt; p[3]       0.042358001 2.525140e-04 0.0158906899  0.017389480  0.040189888</span></span>
<span><span class="co">#&gt;                  97.5%        ESS      Rhat</span></span>
<span><span class="co">#&gt; B[1,2]     0.421482632  496.41018 1.0033982</span></span>
<span><span class="co">#&gt; B[1,3]    -0.086527013  311.31244 1.0034254</span></span>
<span><span class="co">#&gt; B[2,3]     0.928726204   60.24373 1.0086724</span></span>
<span><span class="co">#&gt; sigma      0.702160361 4455.55279 1.0009391</span></span>
<span><span class="co">#&gt; sigma1[1]  0.036117189 1364.89387 1.0042282</span></span>
<span><span class="co">#&gt; sigma1[2]  0.044571476 2874.80598 0.9998832</span></span>
<span><span class="co">#&gt; sigma1[3]  0.069936393 1211.18638 1.0008464</span></span>
<span><span class="co">#&gt; sigma0[1]  0.006155495  178.22450 1.0449098</span></span>
<span><span class="co">#&gt; sigma0[2]  0.009872035  364.44844 1.0050532</span></span>
<span><span class="co">#&gt; sigma0[3]  0.013334242  371.98046 1.0006414</span></span>
<span><span class="co">#&gt; p[1]       0.173356366  648.51063 1.0095333</span></span>
<span><span class="co">#&gt; p[2]       0.070259503  779.62068 1.0019857</span></span>
<span><span class="co">#&gt; p[3]       0.079158536 1734.45765 1.0011530</span></span></code></pre></div>
<p>In this summary, we provide the estimated posterior mean, variance,
standard deviation, 2.5% quantile, median, 97.5% quantile, effective
sample size (ESS), and Gelman-Rubin Rhat statistic for each parameter.
The ESS quantifies the number of independent samples, while Rhat (the
Gelman-Rubin statistic) assesses convergence. See <a href="https://mc-stan.org/docs/2_18/reference-manual/effective-sample-size-section.html" class="external-link">ESS</a>
and <a href="https://mc-stan.org/docs/2_18/reference-manual/notation-for-samples-chains-and-draws.html#split-r-hat-for-detecting-non-stationarity" class="external-link">Rhat</a>
for a detailed explanations of ESS and Rhat.</p>
<p>For the meaning of each parameter, <code>B[l, k]</code> represents
the direct effect of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>-th
latest trait on the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>-th
latest trait. For example: <code>B[1, 2]</code> represents the direct
effect of adult BMI on breast cancer; <code>B[1, 3]</code> represents
the direct direct effect of childhood BMI on breast cancer;
<code>B[2, 3]</code> represents the direct direct effect of childhood
BMI on adult BMI.</p>
<p>See the figure below for a visualization.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/include_graphics.html" class="external-link">include_graphics</a></span><span class="op">(</span><span class="st">"https://zixuanwu1.github.io/FLOW-MR/articles/FLOW-MR-tutorial_files/figure-html/EffectVisual.png"</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://zixuanwu1.github.io/FLOW-MR/articles/FLOW-MR-tutorial_files/figure-html/EffectVisual.png"><!-- --></p>
<p>For the meaning of other parameters, we assume the following priors
in our manuscript:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>α</mi><mrow><mi>k</mi><mi>j</mi></mrow></msub><mo>∼</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><msub><mi>p</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mi>N</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><msubsup><mi>σ</mi><mrow><mi>k</mi><mn>0</mn></mrow><mn>2</mn></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>p</mi><mi>k</mi></msub><mi>N</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><msubsup><mi>σ</mi><mrow><mi>k</mi><mn>1</mn></mrow><mn>2</mn></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">
\alpha_{kj} \sim (1 - p_k)N(0, \sigma_{k0}^2) + p_k N(0, \sigma_{k1}^2)
</annotation></semantics></math></p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>β</mi><mrow><mi>k</mi><mi>l</mi></mrow></msub><mo>∼</mo><mi>N</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo>,</mo><msup><mi>σ</mi><mn>2</mn></msup><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo><mspace width="1.0em"></mspace><mi>l</mi><mo>&gt;</mo><mi>k</mi></mrow><annotation encoding="application/x-tex">
\beta_{kl} \sim N(0, \sigma^2), \quad l &gt; k
</annotation></semantics></math></p>
<p>We also include the posterior summary of these parameters. In the
summary, <code>sigma1[k]</code> represents the standard deviation of the
slab component for trait
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
<code>sigma0[k]</code> represents the standard deviation of the spike
component for trait
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
<code>p[k]</code> represents the proportion of the slab component, and
<code>sigma</code> represents the standard deviation of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>B</mi><annotation encoding="application/x-tex">B</annotation></semantics></math>.</p>
<p>The effects should be considered as significant if its credible
interval does not include 0. For example, here <code>B[2, 3]</code> and
<code>B[1, 3]</code> are found to be significant, while
<code>B[1,2]</code> is not.</p>
</div>
<div class="section level3">
<h3 id="step-4-optional-working-with-raw-outputs">Step 4 (Optional) Working with Raw outputs<a class="anchor" aria-label="anchor" href="#step-4-optional-working-with-raw-outputs"></a>
</h3>
<p>In most cases, the summary table provides sufficient information.
However, in certain situations, it may be useful to directly access the
posterior samples, such as when you need to obtain other quantiles. In
these cases, you can follow the procedure outlined below.</p>
<p>In the Gibbs sampler, we ran four chains in parallel. To access
information from the i-th chain, you can use the command
<code>result$raw[[i]]</code>. For example <code>result$raw[[1]]$B</code>
is a three dimensional array of dimension K by K by N, where K is the
number of traits and N is the number of iterations. Similarly,
<code>result$raw[[1]]$sigma1</code>,
<code>result$raw[[1]]$sigma0</code>, <code>result$raw[[1]]$p</code> are
two dimensional arrays of dimension K by N where K is the number of
traits and N is the number of iterations.</p>
<p>To simplify the procedure, one can use the function
<code>get_posterior_samples</code> in <code>FLOWMR</code> to extract
posterior samples from all four chains for a particular parameter after
a user-defined warm-up period. The parameter must be one of “B”,
“sigma1”, “sigma0”, “sigma”, “p”.</p>
<p>For example, here we extract the posterior samples for
<code>B[1,3]</code> after 3000 warm-up periods</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_B</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_posterior_samples.html">get_posterior_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"B"</span>, K <span class="op">=</span> <span class="fl">3</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span>, warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span></code></pre></div>
<p>We can check its quantiles:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">post_B</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.025</span>, <span class="fl">.5</span>, <span class="fl">0.975</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;        2.5%         50%       97.5% </span></span>
<span><span class="co">#&gt; -0.77297080 -0.40858112 -0.08652701</span></span></code></pre></div>
<p>And we can also draw a histogram of it for visualization:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">post_B</span>, breaks <span class="op">=</span> <span class="fl">30</span>, main <span class="op">=</span> <span class="st">"Histogram of posterior samples of direct effects of X3 on X1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>For <code>sigma1</code>, <code>sgima0</code> and <code>p</code>, the
index argument have to be 1 dimensional. For example</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_p</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_posterior_samples.html">get_posterior_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"p"</span>, K <span class="op">=</span> <span class="fl">3</span>, ind <span class="op">=</span> <span class="fl">1</span>, warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">post_p</span>,  breaks <span class="op">=</span> <span class="fl">30</span>, main <span class="op">=</span> <span class="st">"Histogram of posterior samples of p1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>For <code>sigma</code>, the index argument should be NULL:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_sigma</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_posterior_samples.html">get_posterior_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"sigma"</span>, K <span class="op">=</span> <span class="fl">3</span>, ind <span class="op">=</span> <span class="cn">NULL</span>, warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">post_sigma</span>,  breaks <span class="op">=</span> <span class="fl">30</span>, main <span class="op">=</span> <span class="st">"Histogram of posterior samples of sigma"</span><span class="op">)</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="step-5-optional-mcmc-diagnoistics-">Step 5 (Optional) MCMC diagnoistics.<a class="anchor" aria-label="anchor" href="#step-5-optional-mcmc-diagnoistics-"></a>
</h3>
<p>If non-convergence issues arise (e.g., an Rhat value greater than
1.1), you can generate a trace plot of the posterior samples using the
function <code>traceplot</code>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">FLOWMR</span><span class="fu">::</span><span class="fu"><a href="../reference/traceplot.html">traceplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"B"</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="co"># for B</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">FLOWMR</span><span class="fu">::</span><span class="fu"><a href="../reference/traceplot.html">traceplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"sigma1"</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="co"># for sigma1, sigma0, p</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>The traceplot is a useful tool for diagnosing MCMC convergence (See
this in rstan for example <a href="https://mc-stan.org/rstan/reference/stanfit-method-traceplot.html" class="external-link">link</a>.
In the traceplot, each line corresponds to posterior samples from one
MCMC chain. Non-convergence is indicated when the posterior samples from
the four chains do not appear similar.</p>
<p>If convergence issues arise, consider increasing the number of
iterations or adjusting the prior parameters</p>
</div>
<div class="section level3">
<h3 id="step-6-estimating-path-wise-indirect-effects">Step 6 Estimating path-wise / indirect effects<a class="anchor" aria-label="anchor" href="#step-6-estimating-path-wise-indirect-effects"></a>
</h3>
<p>To estimate path-wise effects, use the following command. For
example, the function <code>indirect_effect</code> evaluates the
indirect effect of childhood BMI on breast cancer through adult BMI:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path_effect</span> <span class="op">=</span> <span class="fu"><a href="../reference/indirect_effect.html">indirect_effect</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, K <span class="op">=</span> <span class="fl">3</span>, path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">1</span><span class="op">)</span>,  warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">path_effect</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Mean        2.5%       50%     97.5%</span></span>
<span><span class="co">#&gt; 1 0.1197818 -0.06143416 0.1103945 0.3452347</span></span></code></pre></div>
<p>The <code>path</code> parameter specifies the path you are interested
it. In this case it is 3 -&gt; 2 -&gt; 1.</p>
<p>This can also be done by first extracting the posterior samples of
`B[1, 2]’ and ‘B[2, 3]’, then multiply them together</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post_B1</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_posterior_samples.html">get_posterior_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"B"</span>, K <span class="op">=</span> <span class="fl">3</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="va">post_B2</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_posterior_samples.html">get_posterior_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, par <span class="op">=</span> <span class="st">"B"</span>, K <span class="op">=</span> <span class="fl">3</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>, warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">post_B1</span> <span class="op">*</span> <span class="va">post_B2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.025</span>, <span class="fl">.5</span>, <span class="fl">.975</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;        2.5%         50%       97.5% </span></span>
<span><span class="co">#&gt; -0.06143416  0.11039454  0.34523474</span></span></code></pre></div>
<p>Alternatively, instead of focusing on a specific path, you can
estimate the sum of effects from all non-direct paths. The function
<code>indirect_effect</code> summarizes the posterior samples of all
non-direct path-wise effects. It outputs array <code>df</code> of
dimension K by K by 4, where the first two dimensions correspond to the
dimensions of <code>B</code>, the third dimension contains the mean,
2.5/% quantile, median, 97.5% quantile respectively.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ind_effect</span> <span class="op">=</span> <span class="fu"><a href="../reference/indirect_effect.html">indirect_effect</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, K <span class="op">=</span> <span class="fl">3</span>, path <span class="op">=</span> <span class="st">"all"</span>,  warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ind_effect</span><span class="op">)</span></span>
<span><span class="co">#&gt; , , Mean</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1] [,2]      [,3]</span></span>
<span><span class="co">#&gt; [1,]    0    0 0.1197818</span></span>
<span><span class="co">#&gt; [2,]    0    0 0.0000000</span></span>
<span><span class="co">#&gt; [3,]    0    0 0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; , , Quantile_0.025</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1] [,2]        [,3]</span></span>
<span><span class="co">#&gt; [1,]    0    0 -0.06143416</span></span>
<span><span class="co">#&gt; [2,]    0    0  0.00000000</span></span>
<span><span class="co">#&gt; [3,]    0    0  0.00000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; , , Quantile_0.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1] [,2]      [,3]</span></span>
<span><span class="co">#&gt; [1,]    0    0 0.1103945</span></span>
<span><span class="co">#&gt; [2,]    0    0 0.0000000</span></span>
<span><span class="co">#&gt; [3,]    0    0 0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; , , Quantile_0.975</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1] [,2]      [,3]</span></span>
<span><span class="co">#&gt; [1,]    0    0 0.3452347</span></span>
<span><span class="co">#&gt; [2,]    0    0 0.0000000</span></span>
<span><span class="co">#&gt; [3,]    0    0 0.0000000</span></span></code></pre></div>
<p>We can also examine the posterior inference for the total effects,
which represent the sum of the direct and indirect effects.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tot_effect</span> <span class="op">=</span> <span class="fu"><a href="../reference/total_effect.html">total_effect</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, K <span class="op">=</span> <span class="fl">3</span>, warmup <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">tot_effect</span><span class="op">)</span></span>
<span><span class="co">#&gt; , , Mean</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1]      [,2]       [,3]</span></span>
<span><span class="co">#&gt; [1,]    0 0.1585711 -0.2970388</span></span>
<span><span class="co">#&gt; [2,]    0 0.0000000  0.7299033</span></span>
<span><span class="co">#&gt; [3,]    0 0.0000000  0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; , , Quantile_0.025</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1]        [,2]       [,3]</span></span>
<span><span class="co">#&gt; [1,]    0 -0.08859471 -0.4794140</span></span>
<span><span class="co">#&gt; [2,]    0  0.00000000  0.5644166</span></span>
<span><span class="co">#&gt; [3,]    0  0.00000000  0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; , , Quantile_0.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1]      [,2]       [,3]</span></span>
<span><span class="co">#&gt; [1,]    0 0.1557088 -0.2931256</span></span>
<span><span class="co">#&gt; [2,]    0 0.0000000  0.7261120</span></span>
<span><span class="co">#&gt; [3,]    0 0.0000000  0.0000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; , , Quantile_0.975</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      [,1]      [,2]       [,3]</span></span>
<span><span class="co">#&gt; [1,]    0 0.4214826 -0.1284121</span></span>
<span><span class="co">#&gt; [2,]    0 0.0000000  0.9287262</span></span>
<span><span class="co">#&gt; [3,]    0 0.0000000  0.0000000</span></span></code></pre></div>
<p>Finally, we demonstrate how to extract posterior samples of total,
indirect, and path-wise effects, similar to how we did for the direct
effects.</p>
<p>For each total effect, we use the function
<code>get_total_samples</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tot_samples</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_total_samples.html">get_total_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, <span class="fl">3</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">tot_samples</span>, breaks <span class="op">=</span> <span class="fl">30</span>,  main <span class="op">=</span> <span class="st">"Histogram of posterior samples of total effects of X3 on X1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>For path-wise effect, we use the function
<code>get_indirect_samples</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">path_samples</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_indirect_samples.html">get_indirect_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, <span class="fl">3</span>, path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">path_samples</span>, breaks <span class="op">=</span> <span class="fl">30</span>,  main <span class="op">=</span> <span class="st">"Histogram of posterior samples of the effect of X3 on X1 through X2"</span><span class="op">)</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>For the joint effect of non-direct paths, we still use the function
<code>get_indirect_samples</code>. In this case we need to set
<code>path</code> to be “all” and specify the index of interests. In
this case, since there is only 1 indirect path, the indirect effect will
be identical to the path-wise effect we just examined</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ind_samples</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_indirect_samples.html">get_indirect_samples</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">raw</span>, <span class="fl">3</span>, path <span class="op">=</span> <span class="st">"all"</span>, ind <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">ind_samples</span>,  breaks <span class="op">=</span> <span class="fl">30</span>,  main <span class="op">=</span> <span class="st">"Histogram of posterior samples of indirect effects of X3 on X1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="FLOW-MR-tutorial_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This tutorial introduced the key functionalities of FLOW-MR. For more
details, refer to the package documentation.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zixuan Wu, Ethan Lewis, Qingyuan Zhao, Jingshu Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
